pkg/network/image-pull-policy_test.go

```diff
@@ -0,0 +1,108 @@
+package network
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	v1 "k8s.io/api/core/v1"
+
+	opv1alpha1 "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+)
+
+var _ = Describe("Testing image-pull-policy", func() {
+	Describe("validateImagePullPolicy", func() {
+		Context("when configuration uses invalid policy type", func() {
+			spec := &opv1alpha1.NetworkAddonsConfigSpec{ImagePullPolicy: v1.PullPolicy("BAD")}
+
+			It("should fail", func() {
+				errorList := validateImagePullPolicy(spec)
+				Expect(len(errorList)).To(Equal(1), "validation failed due to an unexpected error: %v", errorList)
+				Expect(errorList[0].Error()).To(Equal("requested imagePullPolicy 'BAD' is not valid"))
+			})
+		})
+
+		Context("when configration uses a valid policy type", func() {
+			spec := &opv1alpha1.NetworkAddonsConfigSpec{ImagePullPolicy: v1.PullAlways}
+
+			It("should pass", func() {
+				errorList := validateImagePullPolicy(spec)
+				Expect(errorList).To(BeEmpty())
+			})
+		})
+	})
+
+	Describe("fillDefaultsImagePullPolicy", func() {
+		Context("when no policy specified", func() {
+			Context("and there was a policy specified in the previous config", func() {
+				new := &opv1alpha1.NetworkAddonsConfigSpec{}
+				prev := &opv1alpha1.NetworkAddonsConfigSpec{ImagePullPolicy: v1.PullAlways}
+
+				It("should successfully pass", func() {
+					errorList := fillDefaultsImagePullPolicy(new, prev)
+					Expect(errorList).To(BeEmpty())
+				})
+
+				It("and fill in the previously defined policy", func() {
+					Expect(new.ImagePullPolicy).To(Equal(v1.PullAlways))
```

Comment:
while we know that default policy is not Always, it's not strictly speaking a guarantee. And if for some reason it would be made into Always then this test case would not be distinguishable from behavior of e.g. filling it the result with default (now Always) value.

Reply:
Probably my bad specifying first the new config and then old. We set PullAlways in `prev`, we don't specify any policy in `new`. There is no defaulting. Or maybe I don't understand your comment.